coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1959,18 +1970,33 @@ func (s *server) completeWorkspaceBuildJob(ctx context.Context, job database.Pro
 			sidebarAppID = uuid.NullUUID{UUID: id, Valid: true}
 		}
 
+		// NOTE(Cian): If stopping a workspace build, we must ensure that the
+		// SidebarAppID corresponds to an extant coder_app. If it does not, the
+		// build will fail with a foreign key constraint violation as workspace_builds.ai_task_sidebar_app_id
+		// references coder_apps.id.
+		// Unfortunately, this behavior will result in the workspace build being
+		// 'silently' not configured as an AI task except for a warning in the logs.
+		// Ideally, we should also accept an app slug as these are less likely to change.
+		if sidebarAppID.Valid && !slices.Contains(knownAppIDs, sidebarAppID.UUID.String()) {
+			s.Logger.Warn(ctx, "workspace build has unknown sidebar app ID",
+				slog.F("workspace_build_id", workspaceBuild.ID),
+				slog.F("sidebar_app_id", sidebarAppID),
+				slog.F("known_app_ids", knownAppIDs),
+			)
+			sidebarAppID = uuid.NullUUID{}
+			hasAITask = false
```

Comment:
I think this will trigger on basically every workspace stop, since generally speaking, stop transitions won't have any agents. So, every time you stop an AI Task it will disappear.

Has anyone tested this out end to end? Am I wrong about this always triggering on stop?

Reply:
It doesn't always trigger on stop, and it sometimes triggers on start transitions too. See https://github.com/coder/coder/issues/18776#issuecomment-3149939770